target/ppc/translate/storage-ctrl-impl.c.inc

```diff
@@ -21,6 +21,198 @@
  * Store Control Instructions
  */
 
+static bool do_slbie(DisasContext *ctx, arg_X_rb *a)
+{
+    REQUIRE_64BIT(ctx);
+    REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+
+#if defined(TARGET_PPC64)
+    gen_helper_slbie(cpu_env, cpu_gpr[a->rb]);
+#else
+    qemu_build_not_reached();
+#endif
```

Comment:
It would be better to put all TARGET_PPC64 only functions inside a single `#if defined(TARGET_PPC64)`.
This is also closer to the original code, that used the `GEN_HANDLER` macros inside a `#if defined(TARGET_PPC64)`.

Reply:
`#if defined(TARGET_PPC64)` doesn't seem to work outside the functions in decodetree.

Reply:
So, there is no way to not build these functions for 32-bit?
If that's the case, then maybe it would be better to use `REQUIRE_64BIT()`

Reply:
Ah, I see, even with `REQUIRE_64BIT()`, you still need to guard the call to the helper, because it's only defined for 64-bit, right?

Reply:
Yep! That's right